### DISCUSSION TITLE
I was working on my JavaScript code for my website and ran into an error.

### DISCUSSION THREAD

**Discussion by ghost** [community/community#56240] [Programming Help] [answered]
### Body

It says, “ TypeError: null is not an object (evaluating 'ans.value')” when you only check one checkbox
My code is here, https://github.com/pytutorials/pytutorials.github.io/blob/main/quiz.js
https://github.com/pytutorials/pytutorials.github.io/blob/main/challenges/quiz.html

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by lennis-dev** [comment, marked as answer]
Hey @pytutorials 

I think the error is that you are using name="check" for the first two questions and id="check" for the lower ones, so you should change your JavaScript code to:
```javascript
var answers_correct = 0;
var answers_wrong = 0;
if(answers_wrong == 3){
   alert("Try again next time!");
   window.location.assign("https://pytutorials.github.io/");
}
function checkbox_checked(){
  let ans = document.querySelector("input[name=check]:checked");
  try{
    if(!ans){
      throw "Check a checkbox!";
    }else{
      check_answer();
    }
  }catch(e){
     var h1 = document.createElement("h1");
      h1.innerText = e;
      document.body.appendChild(h1);
  }
}
function check_answer(){
    let ans = document.querySelector("input[name=check]:checked");
    if (ans.value != 4) {
      var h1 = document.createElement("h1");
      h1.innerText = "Try Again!";
      h1.id = "header";
      document.body.appendChild(h1);
      answers_wrong = answers_wrong+1;
    }
    if (ans.value == 4) {
      var h1 = document.createElement("h1");
       h1.innerText = "Correct!";
       h1.id = "header";
       document.body.appendChild(h1);
       answers_correct = answers_correct+1;
    }
}
```

I hope this fixes the problem and a tip for the future for your quiz you should use [HTML radios](https://www.w3schools.com/tags/att_input_type_radio.asp).

Best regards
Lennis

**Reply by ghost** [reply]
That didn’t work

**Reply by ghost** [reply]
I think it’s the error handling because now it always says, “Check all boxes” even though I did

**Reply by ghost** [reply]
actually you were right. Thanks!
